src/libais/ais.h

```diff
@@ -2037,6 +2037,318 @@ class Ais8_367_25 : public Ais8 {
+  AIS8_367_33_SENSOR_WIND_REPORT_2 = 11,
+  AIS8_367_33_SENSOR_RESERVED_12 = 12,
+  AIS8_367_33_SENSOR_RESERVED_13 = 13,
+  AIS8_367_33_SENSOR_RESERVED_14 = 14,
+  AIS8_367_33_SENSOR_RESERVED_15 = 15,
+};
+
+class Ais8_367_33_SensorReport {
+ public:
+  Ais8_367_33_SensorEnum report_type = AIS8_367_33_SENSOR_ERROR;
+  int utc_day = 0;
+  int utc_hr = 0;
+  int utc_min = 0;
+  int site_id = 0;
+
+  virtual Ais8_367_33_SensorEnum getType() const = 0;
+  virtual ~Ais8_367_33_SensorReport() {}
+};
+
+unique_ptr<Ais8_367_33_SensorReport>
+ais8_367_33_sensor_report_factory(const AisBitset &bs,
+                                  const size_t offset);
+
+class Ais8_367_33_Location : public Ais8_367_33_SensorReport {
+ public:
+  int version = 0;
+  AisPoint position = {0.0, 0.0};
+  int precision = 0;
+  int altitude_raw = 0;  // 0.1 Meter
+  float altitude = 0.0;  // Meters
+  int owner = 0;
+  int timeout = 0;
+  int spare2 = 0;
+
+  Ais8_367_33_Location(const AisBitset &bs, const size_t offset);
+  ~Ais8_367_33_Location() {}
+  Ais8_367_33_SensorEnum getType() const override { return AIS8_367_33_SENSOR_LOCATION; }
+};
+
+class Ais8_367_33_Station : public Ais8_367_33_SensorReport {
+ public:
+  string name = "";
+  int spare2 = 0;
+
+  Ais8_367_33_Station(const AisBitset &bs, const size_t offset);
+  ~Ais8_367_33_Station() {}
+  Ais8_367_33_SensorEnum getType() const override {return AIS8_367_33_SENSOR_STATION;}
+};
+
+class Ais8_367_33_Wind : public Ais8_367_33_SensorReport {
+ public:
+  int wind_speed = 0;  // Knots
+  int wind_gust = 0;  // Knots
+  int wind_dir = 0;  // Degrees
+  int wind_gust_dir = 0;  // Degrees
+  int sensor_type = 0;
+  int wind_forecast = 0;  // Knots
+  int wind_gust_forecast = 0;  // Knots
+  int wind_dir_forecast = 0;  // Degrees
+  int utc_day_forecast = 0;
+  int utc_hour_forecast = 0;
+  int utc_min_forecast = 0;
+  int duration = 0;  // Minutes
+  int spare2 = 0;
+
+  Ais8_367_33_Wind(const AisBitset &bs, const size_t offset);
+  ~Ais8_367_33_Wind() {}
+  Ais8_367_33_SensorEnum getType() const override {return AIS8_367_33_SENSOR_WIND;}
+};
+
+class Ais8_367_33_WaterLevel : public Ais8_367_33_SensorReport {
+ public:
+  int type = 0;
+  int level = 0;  // Centimeters
+  int trend = 0;
+  int vdatum = 0;
+  int sensor_type = 0;
+  int forecast_type = 0;
+  int level_forecast = 0;  // Centimeters
+  int utc_day_forecast = 0;
+  int utc_hour_forecast = 0;
+  int utc_min_forecast = 0;
+  int duration = 0;  // Minutes
+  int spare2 = 0;
+
+  Ais8_367_33_WaterLevel(const AisBitset &bs, const size_t offset);
+  ~Ais8_367_33_WaterLevel() {}
+  Ais8_367_33_SensorEnum getType() const override {return AIS8_367_33_SENSOR_WATER_LEVEL;}
+};
+
+class Ais8_367_33_Curr2D_Current {
+ public:
+  int speed_raw = 0;  // 0.1 Knots
+  float speed = 0.0;  // Knots
+  int dir = 0;  // Degrees
+  int depth = 0;  // Meters
+};
+
+class Ais8_367_33_Curr2D : public Ais8_367_33_SensorReport {
+ public:
+  Ais8_367_33_Curr2D_Current currents[3];
```

Comment:
Prefer std::array here and everywhere else

Reply:
Okay.  I was modeling my changes after AIS8_1_26.  Will make the change.